boa/src/exec/mod.rs

```diff
@@ -477,6 +529,25 @@ impl Interpreter {
             _ => panic!("TypeError: invalid assignment to {}", node),
         }
     }
+
+    /// Check if the `Value` can be converted to an `Object`
+    ///
+    /// The abstract operation `RequireObjectCoercible` takes argument argument.
+    /// It throws an error if argument is a value that cannot be converted to an Object using `ToObject`.
+    /// It is defined by [Table 15][table]
+    ///
+    /// More information:
+    ///  - [ECMAScript reference][spec]
+    ///
+    /// [table]: https://tc39.es/ecma262/#table-14
+    /// [spec]: https://tc39.es/ecma262/#sec-requireobjectcoercible
+    pub fn require_object_coercible<'a>(&mut self, value: &'a Value) -> Result<&'a Value, Value> {
+        if value.is_null_or_undefined() {
+            self.throw_type_error("cannot convert null or undefined to Object")?;
+            unreachable!();
```

Comment:
I would use `expect_err()` too.

Reply:
Hmmm. I think we should have a version that just constructs the error, maybe something like `construct_type_error` this would return a `Value` so we don't have to do this weird thing with `unreachable!()`

So we would do:
```rust
fn some_function(&self) -> Result<AnyNonValueType, Value> {
	// ... logic here ... 
	
	Err(self.construct_type_error("some message")) 
}
```

But when we have a `ResultValue` we use the `throw_type_error` method

What do you think? @Razican

Reply:
Yes, definitely, this looks much nicer :)

Reply:
Should we do this in this PR or in a different one?

Reply:
I guess we can do this in this PR.